No suitable driver found for jdbc:oracle:thin: ...

I have a bit of a problem running utPLSQL-maven-plugin. If I simply add the plugin 

>
                      <plugin>
                        <groupId>org.utplsql</groupId>
                        <artifactId>utplsql-maven-plugin</artifactId>
                        <version>3.1.0-SNAPSHOT</version>
                        <configuration>
                             ...
                        </configuration>
                       </plugin>

Then I get following error:

> Caused by: java.lang.ClassNotFoundException: oracle.sql.ORAData

This makes me think I need to add oracle jdbc driver onto classpath of the plugin. So I added the dependencies:

>
                    <plugin>
                        <groupId>org.utplsql</groupId>
                        <artifactId>utplsql-maven-plugin</artifactId>
                        <version>3.1.0-SNAPSHOT</version>

                          <dependencies>
                            <dependency>
                              <groupId>com.oracle.jdbc</groupId>
                              <artifactId>ojdbc8</artifactId>
                              <version>12.2.0.1</version>
                            </dependency>
                            <dependency>
                              <groupId>com.oracle.jdbc</groupId>
                              <artifactId>orai18n</artifactId>
                              <version>12.2.0.1</version>
                            </dependency>
                          </dependencies>

That helps a bit, but the test run still fails with 

> Caused by: java.sql.SQLException: No suitable driver found for jdbc:oracle:thin: ...

I can't find where in the code (plugin or java-api) do you register the oracle driver class. I can see that in cli project there is Class.forName(), but nothing like it in plugin nor java-api.

What is the right way to register the driver then?

--- a/.gitignore
+++ b/.gitignore
@@ -20,6 +20,7 @@ target/
 .mvn
 log.txt
 **/pom.xml.versionsBackup
+dependency-reduced-pom.xml
 
 # Docs
 doc/

--- a/pom.xml
+++ b/pom.xml
@@ -5,7 +5,7 @@
 
     <groupId>org.utplsql</groupId>
     <artifactId>utplsql-maven-plugin</artifactId>
-    <version>3.1.2-SNAPSHOT</version>
+    <version>3.1.3-SNAPSHOT</version>
     <packaging>maven-plugin</packaging>
 
     <name>utPLSQL Maven Plugin</name>
@@ -49,9 +49,22 @@
         <maven.version>3.5.0</maven.version>
         <java.version>1.8</java.version>
         <powermock.version>1.7.4</powermock.version>
+        <ojdbc.version>12.2.0.1</ojdbc.version>
     </properties>
 
     <dependencies>
+        <dependency>
+            <groupId>com.oracle.jdbc</groupId>
+            <artifactId>ojdbc8</artifactId>
+            <version>${ojdbc.version}</version>
+        </dependency>
+
+        <dependency>
+            <groupId>com.oracle.jdbc</groupId>
+            <artifactId>orai18n</artifactId>
+            <version>${ojdbc.version}</version>
+        </dependency>
+
         <dependency>
             <groupId>org.utplsql</groupId>
             <artifactId>java-api</artifactId>
@@ -143,6 +156,26 @@
                     <source>${java.version}</source>
                     <target>${java.version}</target>
                     <encoding>${project.build.sourceEncoding}</encoding>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-shade-plugin</artifactId>
+                <version>2.1</version>
+                <executions>
+                    <execution>
+                        <phase>package</phase>
+                        <goals>
+                            <goal>shade</goal>
+                        </goals>
+                    </execution>
+                </executions>
+                <configuration>
+                    <artifactSet>
+                        <includes>
+                            <include>org.utplsql:java-api</include>
+                        </includes>
+                    </artifactSet>
                 </configuration>
             </plugin>
             <plugin>
@@ -327,5 +360,24 @@
                 <enabled>true</enabled>
             </snapshots>
         </repository>
+
+        <repository>
+            <id>maven.oracle.com</id>
+            <releases>
+                <enabled>true</enabled>
+            </releases>
+            <snapshots>
+                <enabled>false</enabled>
+            </snapshots>
+            <url>https://maven.oracle.com</url>
+            <layout>default</layout>
+        </repository>
     </repositories>
+
+    <pluginRepositories>
+        <pluginRepository>
+            <id>maven.oracle.com</id>
+            <url>https://maven.oracle.com</url>
+        </pluginRepository>
+    </pluginRepositories>
 </project>

